validator/sawtooth_validator/networking/interconnect.py

```diff
@@ -351,15 +348,18 @@ def send_message(self, msg, connection_id=None):
         :param msg: protobuf validator_pb2.Message
         """
         zmq_identity = None
-        if connection_id is not None and self._connections is not None:
-            if connection_id in self._connections:
-                connection_info = self._connections.get(connection_id)
-                if connection_info.connection_type == \
-                        ConnectionType.ZMQ_IDENTITY:
-                    zmq_identity = connection_info.connection
-            else:
-                LOGGER.debug("Can't send to %s, not in self._connections",
-                             connection_id)
+
+        try:
+            connection_info = self._connections.get(connection_id)
+        except KeyError:
+            LOGGER.debug("Can't send to %s, not in self._connections",
+                         connection_id)
+
+        try:
+            if connection_info.connection_type == ConnectionType.ZMQ_IDENTITY:
```

Comment:
This could probably just be part of the previous `try/exception` block

Reply:
`connection_info = self._connections.get(connection_id)` returns `None` if the id isn't there, so the extra `except` has to be there. 

Changing `self._connections.get(connection_id)` to `self._connections[connection_id]` would allow putting everything in one block. Is there a reason it isn't like that already?